src/libPMacc/include/particles/ParticlesBase.kernel

```diff
@@ -741,13 +741,24 @@ struct KernelBashParticles
             frame,
             FramePtr
         );
+
+        /* `exchangeChunk` is a view to a chunk of the memory in the exchange-
+         * The chunk contains between 0 and `numBashedParticles` particles
+         * and is updated for each frame.
+         */
         PMACC_SMEM(
-            hasMemory,
-            bool
+            exchangeChunk,
+            TileDataBox< T_ExchangeValueType >
         );
+
+        /* flag: define if all particles from the current frame are copied to the
+         * exchange buffer
+         *
+         * `true` if all particles are copied, else `false`
+         */
         PMACC_SMEM(
-            tmpBorder,
-            TileDataBox< T_ExchangeValueType >
+            allParticlesCopied,
+            bool
```

Comment:
Was is really necessary to exchange the position of the `TileDataBox< T_ExchangeValueType >` and the `bool` variable (while also renaming it!) just to confuse the reviewers? :stuck_out_tongue: I spent like ten seconds starring at the code in line 745 wonding why you retyped from `bool` to `TileDataBox< T_ExchangeValueType >`. :roll_eyes: